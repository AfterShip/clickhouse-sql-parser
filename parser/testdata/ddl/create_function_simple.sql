CREATE FUNCTION linear_equation AS (x, k, b) -> k*x + b;