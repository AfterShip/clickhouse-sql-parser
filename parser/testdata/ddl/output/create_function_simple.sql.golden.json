[
  {
    "CreatePos": 0,
    "OrReplace": false,
    "IfNotExists": false,
    "FunctionName": {
      "Name": "linear_equation",
      "QuoteType": 1,
      "NamePos": 16,
      "NameEnd": 31
    },
    "OnCluster": null,
    "Params": {
      "LeftParenPos": 35,
      "RightParenPos": 43,
      "Items": {
        "ListPos": 36,
        "ListEnd": 43,
        "HasDistinct": false,
        "Items": [
          {
            "Expr": {
              "Name": "x",
              "QuoteType": 1,
              "NamePos": 36,
              "NameEnd": 37
            },
            "Alias": null
          },
          {
            "Expr": {
              "Name": "k",
              "QuoteType": 1,
              "NamePos": 39,
              "NameEnd": 40
            },
            "Alias": null
          },
          {
            "Expr": {
              "Name": "b",
              "QuoteType": 1,
              "NamePos": 42,
              "NameEnd": 43
            },
            "Alias": null
          }
        ]
      },
      "ColumnArgList": null
    },
    "Expr": {
      "LeftExpr": {
        "LeftExpr": {
          "Name": "k",
          "QuoteType": 1,
          "NamePos": 48,
          "NameEnd": 49
        },
        "Operation": "*",
        "RightExpr": {
          "Name": "x",
          "QuoteType": 1,
          "NamePos": 50,
          "NameEnd": 51
        },
        "HasGlobal": false,
        "HasNot": false
      },
      "Operation": "+",
      "RightExpr": {
        "Name": "b",
        "QuoteType": 1,
        "NamePos": 54,
        "NameEnd": 55
      },
      "HasGlobal": false,
      "HasNot": false
    }
  }
]